src/m365/teams/commands/cache/cache-remove.ts

```diff
@@ -0,0 +1,162 @@
+import { Cli, Logger } from '../../../../cli';
+import {
+  CommandOption
+} from '../../../../Command';
+import * as util from 'util';
+import AnonymousCommand from '../../../base/AnonymousCommand';
+import GraphCommand from '../../../base/GraphCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends AnonymousCommand {
+  confirm?: boolean;
+}
+
+class TeamsCacheRemoveCommand extends GraphCommand {
+  public get name(): string {
+    return commands.CACHE_REMOVE;
+  }
+
+  public get description(): string {
+    return 'Removes the Microsoft Teams client cache';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.confirm = (!(!args.options.confirm)).toString();
+    return telemetryProps;
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: (err?: any) => void): void {
+    if (args.options.confirm) {
+      this.clearTeamsCache(logger, cb);
+    }
+    else {
+      logger.logToStderr('This command will execute the following steps.');
+      logger.logToStderr('- Stop the Microsoft Teams client.');
+      logger.logToStderr('- Clear the Microsoft Teams cached files.');
+
+      Cli.prompt({
+        type: 'confirm',
+        name: 'continue',
+        default: false,
+        message: `Are you sure you want to clear your Microsoft Teams cache?`
+      }, (result: { continue: boolean }): void => {
+        if (!result.continue) {
+          cb();
+        }
+        else {
+          this.clearTeamsCache(logger, cb);
+        }
+      });
+    }
+  }
+
+  private async clearTeamsCache(logger: Logger, cb: (err?: any) => void): Promise<void> {
+    try {
+      await this.killRunningProcess(logger);
+      await this.removeCachingFiles(logger);      
+    } 
+    catch {
+      logger.logToStderr('Teams cache not found');
+      cb();
+      return;
+    }
+
+    logger.logToStderr('Teams cache cleared!');
+    cb();
+  }
+
+  private async killRunningProcess(logger: Logger): Promise<void> {
+    if (this.verbose) {
+      logger.logToStderr('Stop Teams client');
+    }
+
+    const platform = process.platform;
+    let cmd = '';
+
+    switch (platform) {
+      case 'win32': 
+        cmd = 'taskkill /IM "Teams.exe" /F'; 
+        break;
+      case 'darwin': 
+        cmd = `ps ax | grep MacOS/Teams -m 1 | grep -v grep | awk '{ print $1 }'`; 
+        break;
+    }
+
+    if (this.debug) {
+      logger.logToStderr(cmd);
+    }
+
+    try {
+      const cmdOutput = await this.exec(cmd);
+
+      if (cmdOutput.stdout !== '' && platform === 'darwin') {
+        await this.exec(`process.kill(${cmdOutput.stdout})`);
+      }
+
+      if (this.verbose) {
+        logger.logToStderr('Teams client closed');
+      }
+    }
+    catch { 
+      if (this.verbose) {
+        logger.logToStderr('Teams client isn\'t running');
```

Comment:
Good point! Within the catch I have put an extra check to ensure the teams client isn't running. For all other cases, users will get the error message. I've also applied this to the function `removeCacheFiles`.

``` typescript
if (errorMessage.includes('ERROR: The process "Teams.exe" not found.')) {
  if (this.verbose) {
    logger.logToStderr('Teams client isn\'t running');
  }
} else {
  throw new Error(errorMessage);
}
```

Here you can see that I'm looking specifically at the error message I get when the command on windows can't find `teams.exe`. With this logic, it would also be interesting to make a comparison with the output a mac would send. What do you think of this approach?

Reply:
Let's see if there's a different way about it so that we don't need to rely on string comparison that's different between OSes but also languages. On mac we can see if we found a Teams PID or not, so that's a reliable way to see if Teams is running or not. If we can do the same on Windows, then we'd have a more reliable way of detection.

Reply:
Good point, should we maybe take this into a new issue, so we can do a bit more of research on it. Currently I am having trouble fetching the Teams PID and validating the output.

Reply:
We can use the current approach for now and meanwhile look at alternatives.